gr-zeromq/lib/base_impl.cc

```diff
@@ -67,32 +107,10 @@ base_sink_impl::base_sink_impl(int type,
                                int timeout,
                                bool pass_tags,
                                int hwm,
+                               bool bind,
                                const std::string& key)
-    : base_impl(type, itemsize, vlen, timeout, pass_tags, key)
+    : base_impl(type, itemsize, vlen, address, timeout, pass_tags, hwm, true, bind, key)
 {
-    /* Set high watermark */
-    if (hwm >= 0) {
-#if USE_NEW_CPPZMQ_SET_GET
-        d_socket.set(zmq::sockopt::sndhwm, hwm);
-#else
-#ifdef ZMQ_SNDHWM
-        d_socket.setsockopt(ZMQ_SNDHWM, &hwm, sizeof(hwm));
-#else // major < 3
-        uint64_t tmp = hwm;
-        d_socket.setsockopt(ZMQ_HWM, &tmp, sizeof(tmp));
-#endif
-#endif
-    }
-
-    /* Set ZMQ_LINGER so socket won't infinitely block during teardown */
-#if USE_NEW_CPPZMQ_SET_GET
-    d_socket.set(zmq::sockopt::linger, LINGER_DEFAULT);
-#else
-    d_socket.setsockopt(ZMQ_LINGER, &LINGER_DEFAULT, sizeof(LINGER_DEFAULT));
-#endif
-
-    /* Bind */
-    d_socket.bind(address);
```

Comment:
Alternatively, you could simply leave this code in the sub-classes and replace the explicit "bind" call with a method on the base class, something like:
```c++
void base_impl::link_to_address(std::string const& address, bool bind)
{
    if (bind) d_socket.bind(address);
    else      d_socket.connect(address);
}
```
This is the most minimal change, preserving the integrity of the base class.